src/main/java/com/cdancy/jenkins/rest/JenkinsClient.java

```diff
@@ -49,29 +55,46 @@ public JenkinsClient() {
      * @param endPoint URL of Jenkins instance.
      * @param authentication authentication used to connect to Jenkins instance.
      * @param overrides jclouds Properties to override defaults when creating a new JenkinsApi.
+     * @param modules a list of modules to be passed to the Contextbuilder, e.g. for logging.
      */
     public JenkinsClient(@Nullable final String endPoint,
             @Nullable final JenkinsAuthentication authentication,
-            @Nullable final Properties overrides) {
+            @Nullable final Properties overrides,
+            @Nullable final List<Module> modules) {
         this.endPoint = endPoint != null
                 ? endPoint
                 : JenkinsUtils.inferEndpoint();
         this.credentials = authentication != null
                 ? authentication
                 : JenkinsUtils.inferAuthentication();
         this.overrides = mergeOverrides(overrides);
-        this.jenkinsApi = createApi(this.endPoint, this.credentials, this.overrides);
+        this.modules = addModules(this.credentials, modules);
```

Comment:
Building off of my previous comment: lets not set these here and instead just do all work within the `createApi` method.